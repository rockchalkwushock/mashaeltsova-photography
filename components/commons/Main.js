/**
 * REVIEW
 * This is where the CSS-Grid is being established!
 * FIXME
 * The only reason there is the `<div />` is `<Meta />` will be seen as an
 * element in the grid and take up space. There is likely a way to hide it
 * from the grid. Fix this later to get rid of this errant `<div />`.
 */
import Meta from './Meta'
import { styles } from '../../lib'

const Main = ({ children }) => (
  <div>
    <Meta />
    <main className="grid">
      {children}
      <style jsx>{`
        .grid {
          background-color: ${styles.colors.white};
          display: grid;
          grid-gap: 1em;
          grid-template-columns: repeat(7, 1fr);
          padding: 0 1em;
        }
      `}</style>
    </main>
  </div>
)

export default Main
