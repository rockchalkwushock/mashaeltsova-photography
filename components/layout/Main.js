import PropTypes from 'prop-types'

import { styles } from '../../lib'

const Main = ({ children, className }) => (
  <main className={className}>
    {children}
    <style jsx>{`
      main {
        background-color: ${styles.colors.powder};
      }
      .index {
        display: grid;
        grid-gap: 1em;
        grid-template-areas: 'about about about about'
          'services services services services'
          'gallery gallery gallery gallery' 'booking booking booking booking';
        grid-template-columns: repeat(4, 1fr);
        padding: 1em;
      }
      .gallery {
        display: -webkit-box;
        display: -ms-flexbox;
        display: flex;
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
        -ms-flex-direction: column;
        flex-direction: column;
      }
    `}</style>
  </main>
)

Main.propTypes = {
  // NOTE Not marking isRequired because I need to handle Error State in routing.
  className: PropTypes.string
}

export default Main
