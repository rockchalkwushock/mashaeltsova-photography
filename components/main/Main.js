import { styles } from '../../lib'

const Main = ({ children }) => (
  <main>
    {children}
    <style jsx>{`
      main {
        background-color: ${styles.colors.powder};
        display: grid;
        grid-gap: 1em;
        grid-template-areas: 'about about about about'
          'services services services services'
          'gallery gallery gallery gallery' 'booking booking booking booking';
        grid-template-columns: repeat(4, 1fr);
        padding: 1em;
      }
    `}</style>
  </main>
)

export default Main
