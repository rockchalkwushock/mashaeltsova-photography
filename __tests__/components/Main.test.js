import React from 'react'
import { render } from 'enzyme'

import Main from '../../components/main/Main'

describe('Component: Main', () => {
  test('Renders without exploding', () => {
    const tree = render(<Main />)
    expect(tree).toMatchSnapshot()
  })
})
