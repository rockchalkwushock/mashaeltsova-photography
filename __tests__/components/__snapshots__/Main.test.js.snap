// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Main Renders without exploding 1`] = `
<main
  class="jsx-3311478223"
/>
`;
